File: ui/src/androidTest/java/com/xmartlabs/bigbang/ui/recyclerview/singleitem/CarAdapter.java
package com.xmartlabs.bigbang.ui.recyclerview.singleitem;

import android.support.annotation.NonNull;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.xmartlabs.bigbang.ui.common.recyclerview.SingleItemBaseRecyclerViewAdapter;
import com.xmartlabs.bigbang.ui.common.recyclerview.SingleItemBaseViewHolder;
import com.xmartlabs.bigbang.ui.recyclerview.common.Car;
import com.xmartlabs.bigbang.ui.test.R;

public class CarAdapter extends SingleItemBaseRecyclerViewAdapter<Car, CarAdapter.CarViewHolder> {
  @NonNull
  @Override
  public CarViewHolder onCreateViewHolder(@NonNull ViewGroup parent) {
    return new CarViewHolder(inflateView(parent, R.layout.item_single));
  }

  @Override
  public void getOnBindViewHolder(@NonNull CarViewHolder viewHolder, @NonNull Car item, int position) {
    viewHolder.bindItem(item);
  }

  static final class CarViewHolder extends SingleItemBaseViewHolder<Car> {
    TextView title;

    public CarViewHolder(@NonNull View view) {
      super(view);
      title = (TextView) view.findViewById(android.R.id.title);
    }

    @Override
    public void bindItem(@NonNull Car item) {
      super.bindItem(item);
      title.setText(item.getModel());
    }
  }
}

File: ui/src/androidTest/java/com/xmartlabs/bigbang/ui/recyclerview/singleitem/CarAdapter.kt
package com.xmartlabs.bigbang.ui.recyclerview.singleitem

import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.xmartlabs.bigbang.ui.common.recyclerview.SingleItemBaseRecyclerViewAdapter
import com.xmartlabs.bigbang.ui.common.recyclerview.SingleItemBaseViewHolder
import com.xmartlabs.bigbang.ui.recyclerview.common.Car
import com.xmartlabs.bigbang.ui.test.R

class CarAdapter : SingleItemBaseRecyclerViewAdapter<Car, CarAdapter.CarViewHolder>() {
  override fun onCreateViewHolder(parent: ViewGroup) =
      CarViewHolder(inflateView(parent, R.layout.item_single))

  class CarViewHolder(view: View) : SingleItemBaseViewHolder<Car>(view) {
    var title: TextView = view.findViewById(android.R.id.title)

    override fun bindItem(item: Car) {
      super.bindItem(item)
      title.text = item.model
    }
  }
}
